### DISCUSSION TITLE
Medias custom field

### DISCUSSION THREAD

**Discussion by ConstantMath** [area17/twill#1099] [Q&A]
Hi all,
Does anyone have an idea, or even more appreciated a concrete example of how to add a custom field in the media panel (under the "Edit info" panel for example).
Thanks a lot for your help.

**Comment by pboivin** [comment]
Hi @ConstantMath, this is not quite documented but there is a way to add extra fields per image under "Edit info". You can use the `extraMetadatas` property of the `medias` field:

```
    @formField('medias', [
        'name' => 'cover',
        'label' => 'Cover',
        'extraMetadatas' => [
            [
                'label' => 'BG Color',
                'name' => 'bgcolor',
                'type' => 'text',
            ]
        ],
    ])
```

No migration is needed.

Then, you can access this field in your templates like this: 

```
$bgcolor = $item->imageObject('cover')->getMetadata('bgcolor');
```

I believe `type` is limited to `text` or `checkbox` for now.

Hope this can help!

**Reply by ConstantMath** [reply]
It helps a lot, thank you @pboivin 
I'm going to try this right now

**Reply by circleb** [reply]
Is there a way to do this on a file field?

**Reply by adam-jones-net** [reply]
Does this work with select fields?  I tried a type of _select_ and _multi_select_ but they don't appear
